openhands/llm/llm.py

```diff
@@ -200,7 +205,40 @@ def wrapper(*args, **kwargs):
 
             try:
                 # we don't support streaming here, thus we get a ModelResponse
-                resp: ModelResponse = self._completion_unwrapped(*args, **kwargs)
+                try:
+                    resp: ModelResponse = self._completion_unwrapped(*args, **kwargs)
+                except RateLimitError as e:
+                    # If we have fallback LLMs, try them
+                    if self._fallback_llms:
+                        # Extract wait time from error message
+                        wait_time = None
+                        if 'Please try again in' in str(e):
```

Comment:
On what LLMs does this work? It really seems like something that litellm would support, it does a lot of string matching on a 100+ LLMs. For example, it does this when it matches various exception strings to a single unified exception that it returns to us (like ContextWindowExceeded across many providers)

Last I checked, they do have fallback LLMs. It doesn't (or didn't? maybe they do now?) do this bit - it doesn't read the response (which may also be in part outside the exception, in the usage field or something, depends on LLM provider)

Reply:
Ah yes they do!